elvish-mode.el

```diff
@@ -63,7 +63,7 @@ expected.")
 
 (defconst elvish-keyword-pattern
   (let ((keywords (cons 'or elvish-keywords)))
-    (eval `(rx (group ,keywords) (not word))))
+    (eval `(rx (not (any word ?_ ?-)) (group ,keywords) (not (any word ?_ ?-)))))
```

Comment:
Thinking about it, I'm pretty sure that you can just use `symbol-start` and `symbol-end` around groups.